source/scale/redis.rst

```diff
@@ -0,0 +1,115 @@
+
+Deployment guide
+----------------
+
+Deploying Redis with Mattermost includes the following 2 steps: `setting up the Redis server <#set-up-a-redis-server>`__, and `configuring Redis in Mattermost <#configure-redis-in-mattermost>`__.
+
+Set up a Redis server
+~~~~~~~~~~~~~~~~~~~~~
+
+1. Download and install the latest release of `Redis 7.x <https://redis.io/download/>`_. See the Redis documentation for installation details specific to your operating system.
+
+2. Configure Redis appropriately for your environment. Ensure that Redis is secured and accessible only by trusted systems.
+
+3. For high availability deployments, consider setting up Redis in a cluster configuration for improved reliability and performance.
+
+Configure Redis in Mattermost
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Follow these steps to configure Mattermost to use your Redis server:
+
+1. Go to **System Console > Environment > Cache**.
+2. Set **Cache Type** to ``redis`` to enable the Redis-specific settings.
+3. Set the Redis server connection details:
+
+  a. Enter **Redis Address** for the Redis server you set up earlier (e.g., ``redis.example.com:6379``).
+  b. (Optional) Enter **Redis Password** if your Redis server requires authentication.
+  c. (Optional) Enter **Redis Database** to specify which Redis database to use (-1 by default, which uses Redis's default database).
+
+4. Save your configuration and restart the Mattermost server.
+
+Alternatively, you can configure Redis in the ``config.json`` file:
+
+.. code-block:: json
+
+    "CacheSettings": {
+        "CacheType": "redis",
+        "RedisAddress": "redis.example.com:6379",
+        "RedisPassword": "",
+        "RedisCachePrefix": "",
+        "RedisDB": -1,
+        "DisableClientCache": false
+    }
+
+Deploy with AWS ElastiCache
+---------------------------
+
+For enterprise-scale deployments, AWS ElastiCache for Redis provides a fully managed Redis service that simplifies deployment, operation, and scaling. 
+
+When deploying Mattermost with AWS ElastiCache:
+
+1. Create an ElastiCache for Redis instance, choosing Redis OSS version 7.1 or later.
+2. Based on load testing results, we recommend starting with a ``cache.m7g.2xlarge`` instance for deployments supporting over 100,000 users.
+3. Configure your Mattermost server to connect to the ElastiCache endpoint.
+4. Monitor CPU utilization closely, as Redis is single-threaded and performance bottlenecks are typically CPU-bound rather than memory-bound.
+
+Performance considerations
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using Redis with Mattermost, consider the following performance insights:
+
+- Current load tests indicate that memory usage of Redis is relatively low, with CPU being the primary bottleneck because Redis is single-threaded.
+- When planning your Redis deployment, use CPU utilization as the primary metric for scaling decisions.
+- For extremely large deployments, consider sharding your Redis deployment to distribute the workload.
```

Comment:
Did we use sharding during our 200k test? Mostly trying to understand whether this is something we tried internally before, or if there's an opportunity to explore this in future ceiling tests.

Reply:
No we didn't. Let me actually remove this line until we have done it ourselves.